sonar-batch/src/main/java/org/sonar/batch/scan/filesystem/FileIndexer.java

```diff
@@ -183,6 +188,56 @@ private boolean accept(InputFile inputFile) {
     return true;
   }
 
+  private class IndexFileVisitor implements FileVisitor<Path> {
+    private InputFileBuilder inputFileBuilder;
+    private DefaultModuleFileSystem fileSystem;
+    private Progress status;
+    private Type type;
+
+    IndexFileVisitor(InputFileBuilder inputFileBuilder, DefaultModuleFileSystem fileSystem, Progress status, InputFile.Type type) {
+      this.inputFileBuilder = inputFileBuilder;
+      this.fileSystem = fileSystem;
+      this.status = status;
+      this.type = type;
+    }
+
+    @Override
+    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Add a length check and use "charAt(0)" instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS3027)